packages/x-data-grid/src/hooks/core/pipeProcessing/useGridPipeProcessing.ts

```diff
@@ -8,12 +8,17 @@ import {
 } from './gridPipeProcessingApi';
 import { useGridApiMethod } from '../../utils/useGridApiMethod';
 
-interface GridPipeGroupCache {
-  processors: Map<string, GridPipeProcessor<any> | null>;
+type Cache = {
+  [G in GridPipeProcessorGroup]?: GroupCache;
+};
+
+type GroupCache = {
+  processors: Map<string, GridPipeProcessor<any>>;
+  processorsAsArray: GridPipeProcessor<any>[];
```

Comment:
Small refactor along with the optimization, the signal-to-noise ratio is high with the long names & prefixed everywhere. This is internal, no need to add the grid prefix.